### PR TITLE
[docker] add aptos-node docker compose

### PR DESCRIPTION
**Pull request by sherry-x**
## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

(You must have submitted a [signed CLA](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#contributor-license-agreement) that includes your GitHub handle prior to us accepting and landing your work. Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/aptos-labs/aptos-core/tree/main/developer-docs-site, and link to your PR here.)

### CODE REVIEW

**Comment by JoshLind** [comment]
@sherry-x, any idea what happened to the commits, here?

File: docker/compose/aptos-node/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# This compose file defines an Aptos node deployment.
+# It includes a validator node and a fullnode.
+# To use this docker compose:
+# - Generate keys: <TODO add commands here>
+# - Download genesis.blob and waypoint.txt
+# - Initialize secure-data.json
```
Comment (JoshLind) [inline docker/compose/aptos-node/docker-compose.yaml:6]: We'll need tooling for this, too, right? Unless we start all at a predefined state, e.g., 0s?

File: docker/compose/aptos-node/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+# Additional information:
+# * If you use this compose for different Aptos Networks, you will need remove the db volume first.
```
Comment (JoshLind) [inline docker/compose/aptos-node/docker-compose.yaml:9]: nit: you will need **to** remove...

File: docker/compose/aptos-node/docker-compose.yaml (hunk 1 above)
Comment (sherry-x) [reply docker/compose/aptos-node/docker-compose.yaml:6]: Yes we need the tooling for this too, should add a <TODO> sign here.

File: docker/compose/aptos-node/docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        target: /opt/aptos/genesis/keys.json
+    command: ["/opt/aptos/bin/aptos-node", "-f", "/opt/aptos/etc/fullnode.yaml"]
+    ports:
+      - "6182:6182"
+      - "8080:8080"
+      - "9101:9101"
+    expose:
+      - 6182
+      - 8080
+      - 9101
```
Comment (davidiw) [inline docker/compose/aptos-node/docker-compose.yaml:88]: need to probably make this distinct

**Review by davidiw — CHANGES_REQUESTED** [review]

File: docker/compose/aptos-node/docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+# Additional information:
+# * If you use this compose for different Aptos Networks, you will need to remove the db volume first.
+# * If you would like to use the current Aptos version within this repository, execute the
+#     `build.sh` in `docker/validator` and change the image tag below to aptos_e2e:latest
+# * Validator images can be found at https://hub.docker.com/repository/docker/aptoslab/validator/tags
+
+# Monitoring:
+# If you want to install the monitoring components for your node deployment,
+# you can symlink the ../monitoring folder into this directory.
```
Comment (rustielin) [inline docker/compose/aptos-node/docker-compose.yaml:16]: this is a bit confusing -- can we get around it?

**Review by rustielin — APPROVED** [review]
LGTM gogogogo!

**Comment by davidiw** [comment]
> LGTM gogogogo!

This isn't finished yet, it still is missing steps.

**Comment by rustielin** [comment]
> This isn't finished yet, it still is missing steps.

Agree. Thinking we can release this and iterate on it. This PR has been pending for a while

**Pull request merged**